app/utils/nonsipp/TaskListUtils.scala

```diff
@@ -55,23 +55,32 @@ object TaskListUtils {
     srn: Srn,
     isPsp: Boolean,
     isLinkActive: Boolean,
+    noChangesSincePreviousVersion: Boolean,
     schemeName: String
   ): TaskListSectionViewModel = {
     val prefix = "nonsipp.tasklist.declaration"
 
     TaskListSectionViewModel(
       s"$prefix.title",
       if (isLinkActive) {
-        val psaOrPspDeclarationUrl =
+        val psaOrPspDeclarationUrl = {
           if (isPsp) {
             controllers.nonsipp.declaration.routes.PspDeclarationController.onPageLoad(srn).url
           } else {
             controllers.nonsipp.declaration.routes.PsaDeclarationController.onPageLoad(srn).url
           }
-        LinkMessage(
-          s"$prefix.complete",
-          psaOrPspDeclarationUrl
-        )
+        }
+        if (noChangesSincePreviousVersion) {
+          LinkMessage(
+            s"$prefix.view",
+            psaOrPspDeclarationUrl
```

Comment:
I think this is supposed to go to the Submission Confirmation page, but it looks like it's still going to the Declaration page - is this correct?